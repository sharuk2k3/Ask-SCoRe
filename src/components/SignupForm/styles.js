import {StyleSheet} from "react-native";
import {
  COLOR_PRIMARY,
  COLOR_GRAY_PRIMARY,
  COLOR_GRAY_SECONDARY,
} from "../../config/styles";

export default StyleSheet.create({
  ViewStyle: {
    backgroundColor: COLOR_GRAY_PRIMARY,
    width: "100%",
    height: 310,
    borderRadius: 20,
    alignSelf: "center",
    marginBottom: "15%",
    marginTop: "10%",
    flexDirection: "column",
  },

  ViewStyle1: {
    alignSelf: "center",
    height: 40,
    width: "84%",
    backgroundColor: "white",
    borderRadius: 20,
    marginTop: "10%",
    paddingLeft: 10,
    flexDirection: "row",
  },

  ViewStyle2: {
    alignSelf: "center",
    height: 40,
    width: "84%",
    backgroundColor: "white",
    borderRadius: 20,
    marginTop: 20,
    paddingLeft: 10,
    flexDirection: "row",
  },

  TextStyle1: {
    flex: 1,
    color: "black",
  },

  TextStyle2: {
    flex: 1,
    color: "black",
  },

  TextStyle4: {
    flex: 1,
    alignSelf: "center",
    marginTop: 12,
    color: COLOR_GRAY_SECONDARY,
    fontSize: 20,
    fontWeight: "800",
  },

  TextStyle3: {
    marginTop: 20,
    marginRight: "8%",
    alignSelf: "flex-end",
  },

  ButtonStyle: {
    backgroundColor: COLOR_PRIMARY,
    width: "70%",
    height: "45%",
    borderRadius: 25,
    justifyContent: "flex-end",
  },

  ButtonViewStyle: {
    zIndex: 2,
    marginTop: "28%",
    alignItems: "center",
    position: "absolute",
    width: 270,
    height: 100,
    top: 200,
    left: 15,
  },

  IconStyle: {
    color: "gray",
    marginRight: 10,
    alignSelf: "center",
  },
});
