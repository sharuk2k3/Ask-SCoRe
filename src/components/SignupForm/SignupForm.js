import React from "react";
import {Component} from "react";
import {firebase} from "../../config/config";
import {View, Text, TouchableOpacity, TextInput} from "react-native";
import {Spinner} from "../common";
import Icon from "react-native-vector-icons/Ionicons";
import styles from "./styles";

class SignupForm extends Component {
  state = {
    email: "",
    password: "",
    fullName: "",
    repassword: "",
    error: "",
    loading: false,
  };

  onButtonPress() {
    const {email, password, repassword, fullName} = this.state;
    this.setState({error: "", loading: true});

    if (password !== repassword) {
      alert("Password don't Match!");
      this.setState({loading: false});
      return;
    }

    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(response => {
        firebase.auth().signOut();
        firebase
          .auth()
          .currentUser.sendEmailVerification()
          .then(() => {
            alert("Verification Email Sent. Please Verify your Email");
          });
        const uid = response.user.uid;
        const data = {
          id: uid,
          email,
          fullName,
          isImagePresent: false,
          postNumber: 0,
          adminRights: false,
        };
        const userRef = firebase.firestore().collection("users");
        const userInf = firebase.firestore().collection("userInfo");
        userRef
          .doc(uid)
          .set(data)
          .then(this.onSigninSuccess.bind(this))
          .catch(error => {
            alert(error);
          });

        userInf
          .doc(email)
          .set({
            id: uid,
          })
          .then(() => {
            null;
          })
          .catch(error => {
            alert(error);
          });
      })
      .catch(error => {
        alert(error);
        this.setState({error: "Signin Error", loading: false});
      });
  }
  onSigninSuccess() {
    this.setState({email: "", password: "", loading: false, error: ""});
  }

  renderButton() {
    if (this.state.loading) {
      return <Spinner size="small" />;
    }
    return (
      <TouchableOpacity
        style={styles.ButtonStyle}
        onPress={this.onButtonPress.bind(this)}>
        <Text style={styles.TextStyle4}>SIGNUP</Text>
      </TouchableOpacity>
    );
  }

  onSigninFail() {
    this.setState({error: "Signin Error", loading: false});
  }

  render() {
    return (
      <View>
        <View style={styles.ViewStyle}>
          <View>
            <View style={styles.ViewStyle1}>
              <Icon style={styles.IconStyle} size={20} name={"mail"} />
              <TextInput
                style={styles.TextStyle1}
                placeholder="example@gmail.com"
                placeholderTextColor={"gray"}
                autoCapitalize="none"
                autoCorrect={false}
                value={this.state.email}
                onChangeText={email => this.setState({email})}
              />
            </View>
            <View style={styles.ViewStyle2}>
              <Icon style={styles.IconStyle} size={20} name={"person"} />
              <TextInput
                style={styles.TextStyle2}
                placeholder="Full Name"
                autoCapitalize="none"
                placeholderTextColor={"gray"}
                autoCorrect={false}
                value={this.state.fullName}
                onChangeText={fullName => this.setState({fullName})}
              />
            </View>
            <View style={styles.ViewStyle2}>
              <Icon style={styles.IconStyle} size={20} name={"eye-off"} />
              <TextInput
                style={styles.TextStyle2}
                placeholder="Password"
                autoCapitalize="none"
                placeholderTextColor={"gray"}
                autoCorrect={false}
                secureTextEntry
                value={this.state.password}
                onChangeText={password => this.setState({password})}
              />
            </View>
            <View style={styles.ViewStyle2}>
              <Icon style={styles.IconStyle} size={20} name={"eye-off"} />
              <TextInput
                style={styles.TextStyle2}
                placeholder="Re-Password"
                autoCapitalize="none"
                placeholderTextColor={"gray"}
                autoCorrect={false}
                secureTextEntry
                value={this.state.repassword}
                onChangeText={repassword => this.setState({repassword})}
              />
            </View>
          </View>
          <View style={styles.ButtonViewStyle}>{this.renderButton()}</View>
        </View>
      </View>
    );
  }
}

export default SignupForm;
