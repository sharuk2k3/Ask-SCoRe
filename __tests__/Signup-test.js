import React from "react";
import SignupForm from "../src/components/SignupForm/SignupForm.js";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("render SignupForm Component Correctly", () => {
  const tree = renderer.create(<SignupForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
