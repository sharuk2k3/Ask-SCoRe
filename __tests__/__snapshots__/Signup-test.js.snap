// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SignupForm Component Correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#dbdbdb",
        "borderRadius": 20,
        "flexDirection": "column",
        "height": 310,
        "marginBottom": "15%",
        "marginTop": "10%",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 40,
            "marginTop": "10%",
            "paddingLeft": 10,
            "width": "84%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "alignSelf": "center",
                "color": "gray",
                "marginRight": 10,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="example@gmail.com"
          placeholderTextColor="gray"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "black",
              "flex": 1,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 40,
            "marginTop": 20,
            "paddingLeft": 10,
            "width": "84%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "alignSelf": "center",
                "color": "gray",
                "marginRight": 10,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Full Name"
          placeholderTextColor="gray"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "black",
              "flex": 1,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 40,
            "marginTop": 20,
            "paddingLeft": 10,
            "width": "84%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "alignSelf": "center",
                "color": "gray",
                "marginRight": 10,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="gray"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "color": "black",
              "flex": 1,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 40,
            "marginTop": 20,
            "paddingLeft": 10,
            "width": "84%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "alignSelf": "center",
                "color": "gray",
                "marginRight": 10,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Re-Password"
          placeholderTextColor="gray"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "color": "black",
              "flex": 1,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 100,
          "left": 15,
          "marginTop": "28%",
          "position": "absolute",
          "top": 200,
          "width": 270,
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#51AD28",
            "borderRadius": 25,
            "height": "45%",
            "justifyContent": "flex-end",
            "opacity": 1,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#d4d4d4",
              "flex": 1,
              "fontSize": 20,
              "fontWeight": "800",
              "marginTop": 12,
            }
          }
        >
          SIGNUP
        </Text>
      </View>
    </View>
  </View>
</View>
`;
